locations/dict_parser.py

```diff
@@ -0,0 +1,71 @@
+from locations.items import GeojsonPointItem
+
+
+class DictParser(object):
```

Comment:
I'm ok with including this, but I worry that we'll be endlessly updating/adding if-checks every time we want to use it with a new spider. If that ends up happening, I'd rather see us keep that tedious work in the spider code instead of trying to modify this parser.

Reply:
+1, there's clear value in handling standards-compliant schema markup, but less so in handling e.g. streetAddress, street_address, line1, line, and whoops we also need locationStreet1, street1, addressLine1, address1_line1, and addr1.

Reply:
I don't think `DictParser` should be updated with every new spider, edge cases should be fixed in spider code, not here. We'll probably workout some bumps early on like with the `LinkedDataParser`, but after that it should stay pretty static.

`line` -> `street_address` is clearly a stretch because I was writing the Poundland spider at the same time. I'll remove that. If I was being cleaver I should have looked through existing spiders and pulled out the most common fields, but instead I just did from memory.

I think we could also do some fancy things with camelCase vs snake_case in `get_first_key` like I'm already doing with upper/lower case to avoid the list being repetitive.

Reply:
I agree that street_address and it's many friends is a stretch, if you see my earlier attempt at similar:

https://github.com/alltheplaces/alltheplaces/pull/3667/files#diff-4e15618ba17581552d29b623a4d025d66598a7b51be5ae1bede253c8a9dae551

then you will see that it does not attempt to so that field. Several hundred spiders say that there is good mileage in the other fields though and CJ's comments above still apply.